Hackathon - Barcelona 2023 Groups (#1970)
nf-core/website: src/content/events/2023/hackathon-october-2023.mdx

:::

# Logistics

## Registration

You can register to attend the hackathon in-person or online using the links on the Nextflow Summit website: [https://summit.nextflow.io/register](https://summit.nextflow.io/register)

You can register for the nf-core hackathon or the Nextflow Summit, or both. You're welcome to attend in-person or online.
Attendance online is free, attending in person costs €79 (€100 incl. VAT).

**Registration closes September 9, or when sold out.**

The 2022 Barcelona hackathon did sell out. So be sure to register early to avoid disappointment!

<a class="btn btn-success btn-lg mb-3" href="https://summit.nextflow.io/register/"><i class="fa-solid fa-id-card me-2"></i> Register now</a>

## Schedule

The hackathon will run from **Monday 16th October** to **Wednesday 18th October**.
Please see the [Summit website Hackathon Agenda](https://summit.nextflow.io/barcelona/agenda/hackathon/#events) for details.
Note that there is a [calendar view](https://summit.nextflow.io/barcelona/agenda/calendar/)
with links that you can use to subscribe / add to your personal calendar.

## Venue: Hotel SB Glow

> The in-person hackathon will be taking place at Hotel SB Glow.
> More information about this venue can be found the the [Nextflow Summit website](https://summit.nextflow.io/).
> 
> [👉🏻 Google Maps 🗺️](https://goo.gl/maps/6uKCKkbciLAR5qgSA)

Reviewer: ```suggestion
The in-person hackathon will be taking place at [Hotel SB Glow](https://goo.gl/maps/6uKCKkbciLAR5qgSA).
More information about this venue can be found the the [Nextflow Summit website](https://summit.nextflow.io/).
```

    this is the group for you!

    Suggested projects / teams:

    - Subworkflows nf-core/tools infrastructure
    - Template customisation with feature flags
    - Stripped down/customisable nf-core pipeline template
    - Samplesheets infrastructure
    - New lint tests (especially for modules, eg. stubs)
    - Google megatests
    - nf-validation plugin

</HackathonGroup>

<HackathonGroup title="Nextflow backbone" color="#238636" proj_url="https://github.com/orgs/nf-core/projects/47/views/2" leaders={{
    'bentsherman': 'Ben Sherman',
    'jordeu': 'Jordi Deu-Pons'
}}>
    New for autumn 2023! Branch out of pure nf-core and get involved in core Nextflow.

    Suggested projects / teams:

    - nf-prov plugin (RO-Crate / BioCompute object support)
    - nf-quilt plugin
    - Other Nextflow plugins
    - Nextflow documentation
    - New core features for Nextflow (see GitHub issues)

</HackathonGroup>

# Social activities

During the hackathon, we will have light-hearted fun and games!
Special edition nf-core rubber ducks are up for grabs for the winners!

:::info
We're still working out the details for these events - please check back soon for more information.
:::

> 
> - Evenings: in-person social events in Barcelona, as part of the Nextflow Summit program.
> - In person + online quiz! Test your nf-core knowledge!
> - Scavenger Hunt
>   - Take photos of you completing as many challenges as you can and share them on the channel!
> - Hackathon bingo
>   - Grab your [bingo card](https://nfcore-bingo.web.app/?game=nf-core-hackathon) and join in the fun.
> - Hidden socks in Gather
>   - Take screenshots of as many as you can find!
>   - See the sock hunt information board in Gather for more information.

Reviewer: ```suggestion

**In person:**

- Evening social events in Barcelona, as part of the Nextflow Summit program.

**Hybrid (in-person and online)**

- nf-core quiz! Test your nf-core knowledge!
- Scavenger Hunt
  - Take photos of you completing as many challenges as you can and share them on the channel!
- Hackathon bingo
  - Grab your [bingo card](https://nfcore-bingo.web.app/?game=nf-core-hackathon) and join in the fun.
- Hidden socks in Gather
  - Take screenshots of as many as you can find!
  - See the sock hunt information board in Gather for more information.
```

## Meta groups

These groups are here to support anyone who may need help getting started at the hackathon.
They are not intended for focused task-specific work.

<HackathonGroup title="Beginners" leaders={{
    'mribeirodantas': 'Marcel Ribeiro-Dantas',
    'abhi18av': 'Abhinav Sharma',
    'FranBonath': 'Franziska Bonath'
}}>
    For this hackathon, we will  have no designated beginners group but will instead
    have a "beginners table" **on the first day**.
    You're welcome to join to get help on how to contribute to open-source,
    and to meet other community contribution beginners.

    Please note that attendees are expected to know the basics of building with Nextflow.

</HackathonGroup>

<HackathonGroup
    title="Online"
    leaders={{
        yuukiiwa: 'Yuk Kei',
        louperelo: 'Louisa Perelo',
        abhi18av: 'Abhinav Sharma',
    }}
>
    If you're feeling lost in Gather.town, or just want to hang out with other online attendees, look out for Yuk Kei
    and Louisa. They'll be able to steer you in the right direction and get you started with one of the project groups.
</HackathonGroup>

## Project groups

> <HackathonGroup title="Pipelines" color="#bf4b8a" proj_url="https://github.com/orgs/nf-core/projects/47/views/4" leaders={{
>     'drpatelh': 'Harshil Patel',
>     'adamrtalbot': 'Adam Talbot',
>     'MaxUlysse': 'Maxime Garcia',
>     'Emiller88': 'Edmund Miller',
>     'sateeshperi': 'Sateesh Peri',
> }}>

Reviewer 2: ```suggestion
<HackathonGroup title="Pipelines" color="#bf4b8a" proj_url="https://github.com/orgs/nf-core/projects/47/views/4" leaders={{
    'drpatelh': 'Harshil Patel',
    'adamrtalbot': 'Adam Talbot',
    'MaxUlysse': 'Maxime Garcia',
    'Emiller88': 'Edmund Miller',
    'sateeshperi': 'Sateesh Peri',
}} slack=https://nfcore.slack.com/archives/C05RYK1F4PP>
```

Author: Sorry, should have said - `slack` here is literally the Slack channel name, not the URL. Also needs to be in quotes.

    The pipelines group will focus on any pipeline-related work.
    This could be a DSL2 conversion of a pipeline, a new or improved feature for a pipeline,
    or an entirely new pipeline
    (please let us register a new pipeline already before the hackathon,
    following [our guidelines](https://nf-co.re/docs/contributing/adding_pipelines#before-you-start)).
    These are typically led by the main pipeline developers if they also attend the hackathon.

    Suggested projects / teams:

    - Implementing nf-test within pipelines
    - Adding new functionality to existing pipelines
    - Updating pipelines to use the latest nf-core template, via template syncs
    - Teams formed around work on specific pipelines
    - Azure megatests roll-out in pipelines

</HackathonGroup>

<HackathonGroup title="Modules & Subworkflows" color="#1f6feb" proj_url="https://github.com/orgs/nf-core/projects/47/views/5" leaders={{
    'JoseEspinosa': 'Jose Espinosa-Carrasco',

>     'ggabernet': 'Gisela Gabernet'

Reviewer 2: ```suggestion
    'FriederikeHanssen': 'Friederike Hanssen'
```

> }}>

Reviewer 2: ```suggestion
}} slack=https://nfcore.slack.com/archives/C05RYK38561>
```

    The modules group welcomes anyone who wants to work on current or new modules + subworkflows
    at [nf-core/modules](https://github.com/nf-core/modules).

    Suggested projects / teams:

    - Implementing nf-test for modules
    - Creating new modules, see issue list
    - Stubs for all modules

</HackathonGroup>

<HackathonGroup title="Infrastructure" color="#9e6a03" proj_url="https://github.com/orgs/nf-core/projects/47/views/3" leaders={{
    'mashehu': 'Matthias Hörtenhuber',
    'mirpedrol': 'Júlia Mir Pedrol',
    'nvnieuwk': 'Nicolas Vannieuwkerke'

> }}>

Reviewer 2: ```suggestion
}} slack=https://nfcore.slack.com/archives/C05RYK591QD>
```

    The nf-core website and central tooling are always being improved.
    If you'd like to make a difference to nf-core without writing any Nextflow code,
    this is the group for you!

    Suggested projects / teams:

    - Subworkflows nf-core/tools infrastructure
    - Template customisation with feature flags
    - Stripped down/customisable nf-core pipeline template
    - Samplesheets infrastructure
    - New lint tests (especially for modules, eg. stubs)
    - Google megatests
    - nf-validation plugin

</HackathonGroup>

<HackathonGroup title="Nextflow backbone" color="#238636" proj_url="https://github.com/orgs/nf-core/projects/47/views/2" leaders={{
    'bentsherman': 'Ben Sherman',
    'jordeu': 'Jordi Deu-Pons'

> }}>

Reviewer 2: ```suggestion
}} slack=https://nfcore.slack.com/archives/C05RM1ASWBZ>
```

    New for autumn 2023! Branch out of pure nf-core and get involved in core Nextflow.

    Suggested projects / teams:

    - nf-prov plugin (RO-Crate / BioCompute object support)
    - nf-quilt plugin
    - Other Nextflow plugins
    - Nextflow documentation
    - New core features for Nextflow (see GitHub issues)

</HackathonGroup>

# Social activities

During the hackathon, we will have light-hearted fun and games!
Special edition nf-core rubber ducks are up for grabs for the winners!

:::info
We're still working out the details for these events - please check back soon for more information.
:::

### In person:

-   Evening social events in Barcelona, as part of the Nextflow Summit program.

### Hybrid (in-person and online)

-   nf-core quiz! Test your nf-core knowledge!

> -   Scavenger Hunt

Reviewer 2: ```suggestion
-   Scavenger Hunt: https://nfcore.slack.com/archives/C05RYKSDL93
```
